From 195997c24f9ca8a5bc548525bc85ab12b8b92e3f Mon Sep 17 00:00:00 2001
From: Michael W <baddaemon87@gmail.com>
Date: Mon, 16 Apr 2018 21:05:50 +0200
Subject: [PATCH 5/5] Parts: Fix notification light app hint

* The old code doesn't work when the entry in the database is not
  available, because it will cause parsePackageList() to return false,
  resulting in no execution of the code taking care of displaying the hint
* Move the hint to a new function and call it at the appropriate places
  to display the hint when no entry to the database was ever made

Thanks to @mikeioannina for noticing the missing hint

Change-Id: Ie3fb39cfa7e7b04a3536f98e93e2168d7367961d
---
 .../NotificationLightSettings.java            | 28 +++++++++++--------
 1 file changed, 17 insertions(+), 11 deletions(-)

diff --git a/src/org/lineageos/lineageparts/notificationlight/NotificationLightSettings.java b/src/org/lineageos/lineageparts/notificationlight/NotificationLightSettings.java
index 348db50..4f48580 100644
--- a/src/org/lineageos/lineageparts/notificationlight/NotificationLightSettings.java
+++ b/src/org/lineageos/lineageparts/notificationlight/NotificationLightSettings.java
@@ -282,6 +282,7 @@ public class NotificationLightSettings extends SettingsPreferenceFragment implem
         Context context = getActivity();
 
         if (!parsePackageList()) {
+            maybeDisplayApplicationHint(context);
             return;
         }
 
@@ -309,17 +310,22 @@ public class NotificationLightSettings extends SettingsPreferenceFragment implem
                 }
             }
 
-            /* Display a pref explaining how to add apps */
-            if (mApplicationPrefList.getPreferenceCount() == 0) {
-                String summary = getResources().getString(
-                        R.string.notification_light_no_apps_summary);
-                String useCustom = getResources().getString(
-                        R.string.notification_light_use_custom);
-                Preference pref = new Preference(context);
-                pref.setSummary(String.format(summary, useCustom));
-                pref.setEnabled(false);
-                mApplicationPrefList.addPreference(pref);
-            }
+            maybeDisplayApplicationHint(context);
+        }
+    }
+
+    private void maybeDisplayApplicationHint(Context context)
+    {
+        /* Display a pref explaining how to add apps */
+        if (mApplicationPrefList != null && mApplicationPrefList.getPreferenceCount() == 0) {
+            String summary = getResources().getString(
+                    R.string.notification_light_no_apps_summary);
+            String useCustom = getResources().getString(
+                    R.string.notification_light_use_custom);
+            Preference pref = new Preference(context);
+            pref.setSummary(String.format(summary, useCustom));
+            pref.setEnabled(false);
+            mApplicationPrefList.addPreference(pref);
         }
     }
 
-- 
2.17.0

